use-cases/using-SRIOV-and-kata.md

```diff
@@ -0,0 +1,313 @@
+        Capabilities: [160] Single Root I/O Virtualization (SR-IOV)
+        Capabilities: [1d0] Access Control Services
+        Kernel driver in use: ixgbe
+        Kernel modules: ixgbe
+```
+
+The important line being:
+
+```
+        Capabilities: [160] Single Root I/O Virtualization (SR-IOV)
+```
+
+If your card does not report this capability, then it does not support SR-IOV.
+
+### IOMMU Groups and PCIe Access Control Services
+
+Run the following command to see how the IOMMU groups are setup on your 
+host system. 
+```
+$ find /sys/kernel/iommu_groups/ -type l
+```
+
+The command's output details whether or not your NIC is setup 
+appropriately with respect to PCIe Access Control Services (ACS). If the PCI 
+bridge is within the same IOMMU group as your NIC, it indicates 
+that either your device does not support ACS or your device does not 
+appropriately share this default capability.
+
+If the IOMMU groups are setup properly, you will have the PCI for each ACS-enabled 
+NIC port in its own IOMMU group. If you do not see any output when running the above
+command, then you likely need to update your host's kernel configuration.
+
+For more details, see http://vfio.blogspot.com/2014/08/iommu-groups-inside-and-out.html
+
+### Update the host kernel
+
+
+Depending on your host kernel configuration, you might have to rebuild the 
+kernel. If the following conditions are true, you do not need to rebuild 
+your kernel: 
+
+- `CONFIG_VFIO_IOMMU_TYPE1`, `CONFIG_VFIO`, and `CONFIG_VFIO_PCI` are set in 
+the kernel configuration. Your kernel is built with VFIO support when 
+configurations are set. 
+- `CONFIG_VFIO_NOIOMMU` is disabled in the host kernel configuration.
+
+See the following steps one through three if you need to rebuild the kernel.
+
+The following steps, which are based on the Ubuntu 16.04 distribution, update 
+the SR-IOV host system. If you use a different distribution, make 
+appropriate adjustments to the commands:
+
+1. Grab kernel sources:
+
+   ```
+   $ sudo apt-get install linux-source-4.10.0
+   $ sudo apt-get install linux-headers-4.10.0
+   $ cd /usr/src/linux-source-4.10.0/
+   $ sudo tar -xvf linux-source-4.10.0.tar.bz2
+   $ cd linux-source-4.10.0
+   $ sudo apt-get install libssl-dev
+   ```
+
+2. Examine and update the `config` file if necessary: 
+
+   ```
+   $ sudo cp /boot/config-4.8.0-36-generic .config
+   $ # verify resulting .config does not have NOIOMMU set; ie: `CONFIG_VFIO_NOIOMMU` is not set
+   $ # verify `CONFIG_VFIO_IOMMU_TYPE1`, `CONFIG_VFIO=m` and `CONFIG_VFIO_PCI=m` are set as well.
+   $ sudo make olddefconfig 
+   ```
+
+   You might want to modify the kernel `Makefile` to add a unique identifier 
+   to the `EXTRAVERSION` variable prior to running the make. Including the `EXTRAVERSION` 
+   variable causes the `uname -r` command to indicate that a customized kernel is
+   installed and running.
+
+3. Build and install the kernel:
+
+   ```
+   $ make -j3 #and go get coffee
```

Comment:
Won't this fail as written as the build is being run as non-`root` but the directory and files might be owned by `root`? See comment above about minimising the use of `root` though.